src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const logs = [];
+  let currentState = state;
+
+  for (const action of actions) {
+    let stateCopy = { ...currentState };
+
+    switch (action.type) {
+      case 'addProperties':
+        if ('extraData' in action) {
+          Object.assign(stateCopy, action.extraData);
+        }
+        break;
+
+      case 'removeProperties':
+        if ('keysToRemove' in action) {
+          for (const remove of action.keysToRemove) {
+            delete stateCopy[remove];
+          }
+        }
+        break;
+
+      case 'clear':
+        stateCopy = {};
+        break;
+
+      default:
+        throw new Error('wrong action type');
+    }
+    currentState = stateCopy || currentState;
```

Comment:
you use unexpected way. It is better to use _Object.assign_ or _spread_ operator

Reply:
thank you !